systemd/systemd_test.go

```diff
@@ -391,6 +391,26 @@ func (s *SystemdTestSuite) TestAvailable(c *C) {
 	c.Check(s.argses, DeepEquals, [][]string{{"--version"}})
 }
 
+func (s *SystemdTestSuite) TestVersion(c *C) {
+	s.outs = [][]byte{
+		[]byte("systemd 223\n+PAM\n"),
+		// for two error cases
+		[]byte("foo 223\n+PAM\n"),
+		[]byte("systemd\n+PAM\n"),
```

Comment:
When I run this on my focal system I see:
```
$ systemd --version
systemd 245 (245.4-4ubuntu3)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```
I wonder if we should test with real systemd output? I suspect that the above will break the parsing as there is a extra " " in the first line(?)

Reply:
Oh, that's interesting, thanks! I was using systemd output from core, it didn't have extra value. Will fix.

Reply:
Hm maybe we should be parsing using bufio.Scanner with split set to bufio.ScanWords ? I played with this in the morning but did not propose. In case it's useful:
```
diff --git a/systemd/systemd.go b/systemd/systemd.go
index 759b75949c..0e7e6f51d1 100644
--- a/systemd/systemd.go
+++ b/systemd/systemd.go
@@ -142,18 +142,21 @@ func Version() (int, error) {
        // of features, e.g:
        // systemd 229
        // +PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP ...
-       r := bufio.NewReader(bytes.NewReader(out))
-       s, err := r.ReadString('\n')
-       if err != nil {
-               return 0, fmt.Errorf("cannot read systemd version: %v", err)
-       }
-
-       parts := strings.Split(s, " ")
-       if len(parts) != 2 || (len(parts) == 2 && parts[0] != "systemd") {
-               return 0, fmt.Errorf("cannot parse systemd version: %s", s)
+       r := bufio.NewScanner(bytes.NewReader(out))
+       r.Split(bufio.ScanWords)
+       var verstr string
+       for i := 0; i < 2; i++ {
+               if !r.Scan() {
+                       return 0, fmt.Errorf("cannot read systemd version: %v", r.Err())
+               }
+               s := r.Text()
+               if i == 0 && s != "systemd" {
+                       return 0, fmt.Errorf("cannot parse systemd version: %s", s)
+               }
+               if i == 1 {
+                       verstr = strings.TrimSpace(r.Text())
+               }
        }
-
-       verstr := strings.TrimSuffix(parts[1], "\n")
        ver, err := strconv.Atoi(verstr)
        if err != nil {
                return 0, fmt.Errorf("cannot convert systemd version to number: %s", verstr)
```

Reply:
I poked around a bit in my VMs:
- 14.04: `systemd 204\n+PAM...`
- 16.04: `systemd 229\n+PAM...`
- 18.04: `systemd 237\n+PAM...`
- 19.04: `systemd 240\n+PAM...`
- 19.10: `systemd 242 (242)\n+PAM...`
- 20.04: `systemd 245 (245.4-4ubuntu3)\n+PAM...`

So it appears the extra bits appeard only very recently

Reply:
> Hm maybe we should be parsing using bufio.Scanner with split set to bufio.ScanWords ? I played with this in the morning but did not propose. In case it's useful:
> 
Thanks, I like it, applied!